File: library/src/main/java/io/blackbox_vision/mvphelpers/utils/bugfix/helper/LifecycleCallbacksAdapter.java
package io.blackbox_vision.mvphelpers.utils.bugfix.helper;

import android.annotation.TargetApi;
import android.app.Activity;
import android.app.Application;
import android.os.Bundle;


/** Helper to avoid implementing all lifecycle callback methods. */
@TargetApi(14)
public class LifecycleCallbacksAdapter implements Application.ActivityLifecycleCallbacks {
    @Override
    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {

    }

    @Override
    public void onActivityStarted(Activity activity) {

    }

    @Override
    public void onActivityResumed(Activity activity) {

    }

    @Override
    public void onActivityPaused(Activity activity) {

    }

    @Override
    public void onActivityStopped(Activity activity) {

    }

    @Override
    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {

    }

    @Override
    public void onActivityDestroyed(Activity activity) {

    }
}

File: library/src/main/java/io/blackbox_vision/mvphelpers/utils/bugfix/helper/LifecycleCallbacksAdapter.kt
package io.blackbox_vision.mvphelpers.utils.bugfix.helper

import android.annotation.TargetApi
import android.app.Activity
import android.app.Application
import android.os.Bundle


/** Helper to avoid implementing all lifecycle callback methods.  */
@TargetApi(14)
open class LifecycleCallbacksAdapter : Application.ActivityLifecycleCallbacks {
    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle) {

    }

    override fun onActivityStarted(activity: Activity) {

    }

    override fun onActivityResumed(activity: Activity) {

    }

    override fun onActivityPaused(activity: Activity) {

    }

    override fun onActivityStopped(activity: Activity) {

    }

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {

    }

    override fun onActivityDestroyed(activity: Activity) {

    }
}